docs/en/development/data_structures.md

```diff
@@ -5,4 +5,879 @@ slug: /internals
+
+This structure is serialized into JSON format and stored in the metadata engine.
+
+#### 3.1.2 Counter
+
+Maintains the value of each counter in the system and the start timestamps of some background tasks, specifically
+
+- usedSpace: used capacity of the file system
+- totalInodes: number of used files in the file system
+- nextInode: the next available inode number (in Redis, the maximum inode number currently in use)
+- nextChunk: the next available sliceId (in Redis, the largest sliceId currently in use)
+- nextSession: the maximum sid (sessionID) currently in use
+- nextTrash: the maximum trash inode number currently in use
+- nextCleanupSlices: timestamp of the last check to clean up residual slices
+- lastCleanupSessions: timestamp of the last check to clean up stale sessions
+- lastCleanupFiles: timestamp of the last check to clean up residual files
+- lastCleanupTrash: timestamp of the last check to clean up the trash
+
+#### 3.1.3 Session
+
+Records the session IDs of clients connected to this file system and their timeouts. Each client sends a heartbeat message to update the timeout, and those who have not updated for a long time will be automatically cleaned up by other clients.
+
+:::tip
+Read-only clients cannot write to the metadata engine, so their sessions **will not** be recorded.
+:::
+
+#### 3.1.4 SessionInfo
+
+Records specific metadata about the client session so that it can be viewed with the `juicefs status` command. This is specified as
+
+```go
+type SessionInfo struct {
+	Version    string // JuiceFS version
+	HostName   string // Host name
+	MountPoint string // path to mount point. S3 gateway and WebDAV server are "s3gateway" and "webdav" respectively
+	ProcessID  int    // Process ID
+}
+```
+
+This structure is serialized into JSON format and stored in the metadata engine.
+
+#### 3.1.5 Node
+
+Records information about the attributes of each file, as follows
+
+```go
+type Attr struct {
+	Flags     uint8  // reserved flags
+	Typ       uint8  // type of a node
+	Mode      uint16 // permission mode
+	Uid       uint32 // owner id
+	Gid       uint32 // group id of owner
+	Rdev      uint32 // device number
+	Atime     int64  // last access time
+	Mtime     int64  // last modified time
+	Ctime     int64  // last change time for meta
+	Atimensec uint32 // nanosecond part of atime
+	Mtimensec uint32 // nanosecond part of mtime
+	Ctimensec uint32 // nanosecond part of ctime
+	Nlink     uint32 // number of links (sub-directories or hardlinks)
+	Length    uint64 // length of regular file
+
+	Parent    Ino  // inode of parent; 0 means tracked by parentKey (for hardlinks)
+	Full      bool // the attributes are completed or not
+	KeepCache bool // whether to keep the cached page or not
+}
+```
+
+A few of the fields that need clarification.
+
+- Atime/Atimensec: set only when the file is created and when `SetAttr` is actively called, while accessing and modifying the file normally does not affect the Atime value
```

Comment:
```suggestion
- Atime/Atimensec: can be set only when the file is created and when `SetAttr` is actively called, while accessing and modifying the file usually does not affect the Atime value
```

Reply:
delete `can be`, since the set is passive